### DISCUSSION TITLE
List of Style Options

### DISCUSSION THREAD

**Discussion by mauricioluz** [dicebear/dicebear#204] [Q&A] [answered]
Are there a way to get all style options available to a specific style?

**Comment by FlorianKoerner** [comment, marked as answer]
Hi @mauricioluz 

Each avatar style also includes a [JSONSchema](https://json-schema.org/) that we use ourselves to evaluate the available options.

```ts
import { schema as coreSchema } from '@dicebear/avatars';
import { schema as styleSchema } from '@dicebear/adventurer';

console.log(coreSchema, styleSchema);
```

We use this information in the CLI, for example:

```
dicebear create <avatar-style> --help
```

- https://github.com/dicebear/dicebear/blob/88c396fdffadf8d12f2017aeab10e33c570305f1/packages/dicebear/src/utils/command/makeCreateStyleCommand.ts#L14-L39
- https://github.com/dicebear/dicebear/blob/88c396fdffadf8d12f2017aeab10e33c570305f1/packages/dicebear/src/utils/getOptionsBySchema.ts#L5-L50


And also in the documentation:

- https://avatars.dicebear.com/styles/adventurer#style-options
- https://github.com/dicebear/dicebear/blob/88c396fdffadf8d12f2017aeab10e33c570305f1/website/src/components/docs/style/options.tsx
